dbt/adapters/bigquery/impl.py

```diff
@@ -278,7 +278,8 @@ def poll_until_job_completes(cls, job, timeout):
             raise dbt.exceptions.RuntimeException("BigQuery Timeout Exceeded")
 
         elif job.error_result:
-            raise job.exception()
+            e = job.exception()
+            raise type(e)(message=e.message, errors=job.errors)
```

Comment:
I'm not sure about calling `type` to get a class object and just assuming that it works to call as a constructor. I mean, I know it's ok here, but `job` and `job.exception()` come from google, not us.

Is this interface (in particular, the fact that the `__init__` of the exception class returned by `job.exception()` accepts an `errors` keyword argument) considered stable in any way?

I think I would prefer something like:
```
msg = '{}\n{}'.format(e.message, '\n'.join(str(e) for e in job.errors)).strip()
raise dbt.exceptions.RuntimeException(msg)
```

I haven't tested it, and I'm not 100% sure on the type of `job.errors`, but I assume something like that would work.

Reply:
Yeah, totally fair, I went back and forth on that myself. In the end I decided not to hardcode a dbt exception since I wasn't sure about the implications of that downstream for logging. If you're more comfortable with raising a `RuntimeException` as you outlined, I'll change it.

Google API Errors inherit from a base class (GoogleAPICallError) that accepts `errors` and `message` as keyword arguments, so it should be safe to assume we can pass those args. Alternatively, we could hardcode a generic Google API Error exception (see [here](https://github.com/googleapis/google-cloud-python/blob/master/api_core/google/api_core/exceptions.py#L79)) which would ensure we can pass those args, rather than using `type`.

What do you think?

Reply:
Unless it has a negative downstream impact (triggering the exception handler in the wrong way, comes to mind) I would prefer to raise a dbt-native exception. At some point we'll convert it anyway for display, might as well get it done early.